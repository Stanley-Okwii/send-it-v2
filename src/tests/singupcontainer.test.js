import React from "react";
import { shallow} from "enzyme";
import {
  SignUpContainer as App,
  mapStateToProps,
  mapDispatchToProps
} from "../containers/SignUpContainer";

describe("SignUpContainer ", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it("reponds to events", () => {
    const wrapper = shallow(<App />);
    const event = {};
    wrapper.instance().onTextChange(event, { name: "email", value: "email" });
    wrapper.instance().onTextChange(event, { name: "username", value: "dope" });
    wrapper
      .instance()
      .onTextChange(event, { name: "password1", value: "email" });
    wrapper
      .instance()
      .onTextChange(event, { name: "password2", value: "email" });

    expect(wrapper.instance().state.username).toEqual("dope");
  });

  it("reponds to click events", () => {
    const wrapper = shallow(<App register={jest.fn()} />);
    wrapper.find(".index-button").simulate("click");
  });

  it("matches state to props", () => {
    const initialState = {};
    mapStateToProps(initialState);
  });

  it("match dispatch to props", () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).register();
  });
});
