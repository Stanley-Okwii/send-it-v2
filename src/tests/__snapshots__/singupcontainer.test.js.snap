// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpContainer  renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FormField>
          <FormInput
            as={[Function]}
            control={[Function]}
            name="username"
            onChange={[Function]}
            placeholder="username"
            value=""
          />
        </FormField>,
        <FormField>
          <FormInput
            as={[Function]}
            control={[Function]}
            name="email"
            onChange={[Function]}
            placeholder="email"
            value=""
          />
        </FormField>,
        <FormField>
          <FormInput
            as={[Function]}
            control={[Function]}
            name="password1"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
        </FormField>,
        <Button
          as="button"
          className="index-button"
          color="teal"
          onClick={[Function]}
        >
          sign up
        </Button>,
      ],
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormInput
            as={[Function]}
            control={[Function]}
            name="username"
            onChange={[Function]}
            placeholder="username"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": [Function],
            "control": [Function],
            "name": "username",
            "onChange": [Function],
            "placeholder": "username",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormInput
            as={[Function]}
            control={[Function]}
            name="email"
            onChange={[Function]}
            placeholder="email"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": [Function],
            "control": [Function],
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormInput
            as={[Function]}
            control={[Function]}
            name="password1"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": [Function],
            "control": [Function],
            "name": "password1",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "sign up",
          "className": "index-button",
          "color": "teal",
          "loading": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "sign up",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FormField>
            <FormInput
              as={[Function]}
              control={[Function]}
              name="username"
              onChange={[Function]}
              placeholder="username"
              value=""
            />
          </FormField>,
          <FormField>
            <FormInput
              as={[Function]}
              control={[Function]}
              name="email"
              onChange={[Function]}
              placeholder="email"
              value=""
            />
          </FormField>,
          <FormField>
            <FormInput
              as={[Function]}
              control={[Function]}
              name="password1"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
          </FormField>,
          <Button
            as="button"
            className="index-button"
            color="teal"
            onClick={[Function]}
          >
            sign up
          </Button>,
        ],
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormInput
              as={[Function]}
              control={[Function]}
              name="username"
              onChange={[Function]}
              placeholder="username"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "control": [Function],
              "name": "username",
              "onChange": [Function],
              "placeholder": "username",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormInput
              as={[Function]}
              control={[Function]}
              name="email"
              onChange={[Function]}
              placeholder="email"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "control": [Function],
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormInput
              as={[Function]}
              control={[Function]}
              name="password1"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "control": [Function],
              "name": "password1",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "sign up",
            "className": "index-button",
            "color": "teal",
            "loading": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "sign up",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
